src/Dashboard/Dashboard.js

```diff
@@ -27,16 +31,61 @@ const useStyles = makeStyles(theme => ({
 
 const Dashboard = (props) => {
   const { t } = useTranslation();
-
   const classes = useStyles();
+  const {
+    iframeTitle,
+    url,
+    scenarioName,
+    scenarioId,
+    scenarioState,
+    ...otherProps
+  } = props;
+  const formattedUrl = url.replaceAll('<ScenarioName>', scenarioName).replaceAll('<ScenarioId>', scenarioId);
 
-  const { iframeTitle, url, scenarioName, scenarioId, ...otherProps } = props;
+  const getPlaceHolder = (label_key, default_label, icon) => {
```

Comment:
This could be placed in an Outter Component (named DashBoardPlaceholder or something else) instead of a inner function